File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        unordered_map<string, vector<string>> mp;
        
        for(int i=0; i<strs.size(); i++){
            string st = strs[i];
            sort(st.begin(), st.end());
            mp[st].push_back(strs[i]);
        }

        for(auto& [_, vec]: mp){
            result.push_back(vec);
        }

        return result;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        from collections import defaultdict

        anagrams = defaultdict(list)

        for st in strs:
            key = ''.join(sorted(st))
            anagrams[key].append(st)
        
        return list(anagrams.values())